src/Raven.Server/Documents/Handlers/Debugging/StorageHandler.cs

```diff
@@ -345,49 +345,46 @@ public async Task ScratchBufferPoolInfoReport()
             var name = GetStringQueryString("name");
             var typeAsString = GetStringQueryString("type", false);
 
-            IEnumerable<StorageEnvironmentWithType> envs;
-
             if (typeAsString != null)
             {
                 if (Enum.TryParse(typeAsString, out StorageEnvironmentWithType.StorageEnvironmentType type) == false)
                     throw new InvalidOperationException("Query string value 'type' is not a valid environment type: " + typeAsString);
                 var db = Database.GetAllStoragesEnvironment()
                     .FirstOrDefault(x => string.Equals(x.Name, name, StringComparison.OrdinalIgnoreCase) && x.Type == type);
-                envs = db == null ? null : new List<StorageEnvironmentWithType>() { db };
-            }
-            else
-            {
-                envs = Database.GetAllStoragesEnvironment();
-            }
 
-            if (envs == null)
-            {
-                HttpContext.Response.StatusCode = (int)HttpStatusCode.NotFound;
-                return;
+                using (ContextPool.AllocateOperationContext(out DocumentsOperationContext context))
```

Comment:
https://github.com/ravendb/ravendb/pull/18055 returns 404 in this case, also old code. What it should be?

Reply:
for v6.0 https://github.com/ravendb/ravendb/pull/18072